src/pages/index.js

```diff
@@ -74,43 +150,136 @@ previewModalCloseBtn.addEventListener("click", () => {
   closeModal(previewModal);
 });
 
-function getCardElement(data) {
-  const cardElement = cardTemplate.content
-    .querySelector(".card")
-    .cloneNode(true);
-  const cardNameElement = cardElement.querySelector(".card__title");
-  const cardImageElement = cardElement.querySelector(".card__image");
-  const cardLikedBtn = cardElement.querySelector(".card__like-btn");
-  const deleteButton = cardElement.querySelector(".card__delete-btn");
+let selectedCard;
+let selectedCardId;
+const deleteForm = document.querySelector("#delete-form");
 
-  cardImageElement.src = data.link;
-  cardImageElement.alt = data.name;
-  cardNameElement.textContent = data.name;
+function getCardElement(cardData) {
+  const cardTemplate = document.querySelector("#card-template").content;
+  const cardElement = cardTemplate.querySelector(".card").cloneNode(true);
 
-  cardLikedBtn.addEventListener("click", () => {
-    cardLikedBtn.classList.toggle("card__like-btn_liked");
-  });
+  // Add a unique ID to the card
+  cardElement.id = `card-${cardData._id}`;
+
+  const cardImage = cardElement.querySelector(".card__image");
+  const cardTitle = cardElement.querySelector(".card__title");
+  const cardLikeButton = cardElement.querySelector(".card__like-btn");
+  const deleteButton = cardElement.querySelector(".card__delete-btn");
+
+  cardImage.src = cardData.link;
+  cardImage.alt = cardData.name;
+  cardTitle.textContent = cardData.name;
 
   deleteButton.addEventListener("click", () => {
-    cardElement.remove();
+    handleDeleteCard(cardElement, cardData);
   });
 
-  cardImageElement.addEventListener("click", () => {
-    openModal(previewModal);
+  // Set initial like state
+  if (cardData.isLiked) {
+    cardLikeButton.classList.add("card__like-btn_liked");
+  }
 
-    previewModalImageElement.src = data.link;
-    previewModalImageElement.alt = data.name;
-    previewModalCaptionElement.textContent = data.name;
+  // Add click event listener for like button
+  cardLikeButton.addEventListener("click", () => {
+    handleLikeClick(cardData._id, cardLikeButton);
   });
-
   return cardElement;
 }
 
+// Open Avatar Modal
+profileAvatarBtn.addEventListener("click", () => {
+  resetValidation(avatarForm, settings);
+  openModal(avatarModal);
+});
+
+// Close Avatar Modal
+avatarModalCloseBtn.addEventListener("click", () => closeModal(avatarModal));
+
+// Avatar Form Submission
+avatarForm.addEventListener("submit", handleAvatarSubmit);
+
 function openModal(modal) {
   modal.classList.add("modal_opened");
   document.addEventListener("keydown", handleEscapeKey);
 }
 
+// Avatar
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = avatarForm.querySelector(".modal__submit-btn");
+  const avatarImg = document.querySelector(".profile__avatar");
```

Comment:
`document.querySelector(".profile__avatar")` 

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine

Reply:
My remark hasn't been corrected